src/lib.rs

```diff
@@ -27,39 +30,99 @@ pub mod user;
+            Self::Deserialize => "Server was unable to deserialize the data. Try again later.",
+            Self::Directory => {
+                "Server is missing a directory and could not create it. Try again later."
+            }
+            Self::GroupNotFound => "Group not found.",
+            Self::HubNotFound => "Hub not found.",
+            Self::MemberNotFound => "Hub member not found.",
+            Self::Muted => "You are muted.",
+            Self::NoPermission => "You do not have permission to do that.",
+            Self::NotInHub => "You are not in that hub.",
+            Self::ReadFile => "Server was unable to read the data. Try again later.",
+            Self::Serialize => "Server was unable to serialize the data. Try again later.",
+            Self::UserNotFound => "User not found.",
+            Self::WriteFile => "Server was unable to store the data. Try again later.",
+            Self::DeleteFailed => "Server was unable to delete the data.",
+            Self::UnexpectedServerArg => "Something strange happened...",
+            Self::MessageTooBig => "Message too big.",
+            Self::InvalidMessage => "Messages must be sent as UTF-8 strings.",
+            Self::Other(message, _) => message,
+        }
+    }
+
+    fn http_status_code(&self) -> StatusCode {
+        match self {
+            Self::BadAuth => StatusCode::UNAUTHORIZED,
+            Self::InvalidName => StatusCode::BAD_REQUEST,
+            Self::Banned => StatusCode::FORBIDDEN,
+            Self::ChannelNotFound => StatusCode::NOT_FOUND,
+            Self::Deserialize => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::Directory => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::GroupNotFound => StatusCode::NOT_FOUND,
+            Self::HubNotFound => StatusCode::NOT_FOUND,
+            Self::MemberNotFound => StatusCode::NOT_FOUND,
+            Self::Muted => StatusCode::FORBIDDEN,
+            Self::NoPermission => StatusCode::FORBIDDEN,
+            Self::NotInHub => StatusCode::NOT_FOUND,
+            Self::ReadFile => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::Serialize => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::UserNotFound => StatusCode::NOT_FOUND,
+            Self::WriteFile => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::DeleteFailed => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::UnexpectedServerArg => StatusCode::INTERNAL_SERVER_ERROR,
+            Self::MessageTooBig => StatusCode::BAD_REQUEST,
+            Self::InvalidMessage => StatusCode::BAD_REQUEST,
+            Self::Other(_, code) => code.clone(),
+        }
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.to_string().as_str())
+    }
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
 pub static USER_AGENT_STRING: &str = concat!("WICRS Server ", env!("CARGO_PKG_VERSION"));
```

Comment:
There is nothing preventing this to be `const`. But do some profiling because `const` values are inlined whereas `static` values is just one memory location. See [this](https://stackoverflow.com/questions/52751597/what-is-the-difference-between-a-constant-and-a-static-variable-and-which-should).